### PR TITLE
Remove CPhase gate alias references

### PR DESCRIPTION
**Pull request by grwlf**
**Context:** Transition to the quantum device config schema 2

**Description of the Change:**

* Remove mentions of CPhase "gate" which happens to be a Python alias for ControlledPhaseShift.
* Remove the Python-only Projector patch because we already have a relaxed validation procedure for observables
* Relax operation validation by allowing kokkos device to declare C(GlobalPhase) which is a Python-only operation.

**Benefits:**
**Possible Drawbacks:** 
**Related GitHub Issues:**

* https://github.com/PennyLaneAI/pennylane-lightning/pull/642
  - (related discussion) https://github.com/PennyLaneAI/pennylane-lightning/pull/642#discussion_r1535478642

### CODE REVIEW

**Comment by dime10** [comment]
This can be closed right?

**Comment by grwlf** [comment]
> ~This can be closed right?~ My bad I thought this was just about the `CPhase` gate.

Even for CPhase, we need to remove it from tomls contained in the Catalyst repo too.

File: frontend/catalyst/utils/runtime.py
Comment (dime10) [inline frontend/catalyst/utils/runtime.py:129]: I don't like these special cases to be honest. There is a cost/benefit analysis we have to make here. Ultimately what the Python device states is irrelevant, the toml file is the source of truth for the compiler and it will decompose accordingly.
The verification that the set of toml gates matches the set of python device gates was merely added as a nice to have so that it wouldn't go unnoticed if a new gate is added to the python but not the toml file. One could even argue that we shouldn't have this verification at all since the Python device declarations are not relevant to the compiler.

Now recently this internal verification (for our developers) was made stricter and this is leading to problems like having to hardcode special cases. I would rather lean towards removing or relaxing it rather than working in contortions around the rules.

Comment (dime10) [reply frontend/catalyst/utils/runtime.py:129]: > What do you think about removing the validation simultaneously with removing the old device api?

Alright I could be persuaded of that 😅

File: frontend/catalyst/utils/toml.py
Comment (dime10) [inline frontend/catalyst/utils/toml.py:306]: We're about to merge 642 so let's remove these 3 blocks with TODOs

File: frontend/catalyst/utils/runtime.py
Comment (dime10) [inline frontend/catalyst/utils/runtime.py:132]:
```suggestion
```

File: frontend/catalyst/utils/runtime.py
Comment (dime10) [inline frontend/catalyst/utils/runtime.py:126]:
```suggestion
            # TODO: reconsider the TOML-PLDevice verification (with the goal to eliminate this special
            #       case) upon removal of TOML schema 1 support
            # Lightning-kokkos device supports GlobalPhase via PythonAPI but does not via C++ API.
```

**Review by erick-xanadu — CHANGES_REQUESTED** [review]
After #745 is merged and this PR is updated, one should run latest/latest/latest on this branch. I expect it to succeed. On #745 [latest/latest/latest fails](https://github.com/PennyLaneAI/catalyst/actions/runs/9098344802/job/25008333441) with the error message: 

```
FAILED frontend/test/pytest/test_transform.py::TestBroadcastExpand::test_expansion_qnode_no_cache[lightning.kokkos-obs1-params2] - catalyst.utils.exceptions.CompileError: Gates in qml.device.operations and specification file do not match.
Gates that present only in the device: {'C(GlobalPhase)'}
Gates that present only in spec: set()
```

I am not asking for this to fix latest/latest/latest but merely to report what the outcome is on the channel so that we can fix it if this is not the case.

**Review by erick-xanadu — APPROVED** [review]
Since PR #745 might interfere with this PR I am just going to approve it now. But I will leave it to @dime10 to continue this review. I don't want to block this unnecessarily long. Cheers!

File: frontend/catalyst/utils/toml.py
Comment (grwlf) [reply frontend/catalyst/utils/toml.py]: No more relevant

File: frontend/catalyst/utils/runtime.py
Comment (grwlf) [reply frontend/catalyst/utils/runtime.py]: No more relevant

**Comment by grwlf** [comment]
> After #745 is merged and this PR is updated, one should run latest/latest/latest on this branch. I expect it to succeed. On #745 [latest/latest/latest fails](https://github.com/PennyLaneAI/catalyst/actions/runs/9098344802/job/25008333441) with the error message:
> 
> ```
> FAILED frontend/test/pytest/test_transform.py::TestBroadcastExpand::test_expansion_qnode_no_cache[lightning.kokkos-obs1-params2] - catalyst.utils.exceptions.CompileError: Gates in qml.device.operations and specification file do not match.
> Gates that present only in the device: {'C(GlobalPhase)'}
> Gates that present only in spec: set()
> ```
> 
> I am not asking for this to fix latest/latest/latest but merely to report what the outcome is on the channel so that we can fix it if this is not the case.

@erick-xanadu I merged this PR with the recent main and solved the conflicts. The test you mentioned should pass now.

File: frontend/catalyst/utils/runtime.py
Comment (grwlf) [reply frontend/catalyst/utils/runtime.py]: Outdates. This change does not exist any more.

File: frontend/catalyst/utils/toml.py
Comment (dime10) [inline frontend/catalyst/utils/toml.py:276]: Why is only the comment removed instead of the code block? Wasn't that the intention of the todo to remove the code block?

File: frontend/catalyst/utils/toml.py
Comment (dime10) [inline frontend/catalyst/utils/toml.py:272]: Same here

File: frontend/catalyst/utils/toml.py
Comment (dime10) [reply frontend/catalyst/utils/toml.py:276]: If the toml is still incorrect for lightning.kokkos there should be another PR to lightning right?

Comment (grwlf) [reply frontend/catalyst/utils/toml.py:276]: Both code fragments are now in the `patch_schema1_collections` function. It gets called for toml schema 1 only [here](https://github.com/PennyLaneAI/catalyst/pull/718/files/7800c69ebcebebf1b65d9fde2f1ba3c4851cf85a#diff-a87ceb7a61721c8cbd1c58d63193282cae87a0c3e8f8ea79c6ebbcde7a4770efR408). If I remember it correctly, both comments were created when this code resided in the common execution path.

File: frontend/catalyst/utils/toml.py
Comment (grwlf) [reply frontend/catalyst/utils/toml.py:272]: See the comment in the same discussion

File: frontend/catalyst/utils/toml.py
Comment (dime10) [reply frontend/catalyst/utils/toml.py:276]: But the fact that we patch this gate into the toml file doesn't that mean that the upstream toml file is out of date?

Comment (dime10) [reply frontend/catalyst/utils/toml.py:276]: Wait these patches are for schema 1 only? Kokkos is already on schema 2, so why not delete these patches?

Comment (grwlf) [reply frontend/catalyst/utils/toml.py:276]: Yes they are. Schema 2 is in PLL master, but `pip install` installs Pennylane-lightning with old schema 1.

Comment (grwlf) [reply frontend/catalyst/utils/toml.py]: Hmm. Actually, this particular patch (adding Projector gate) is really outdated so I am removing it. Schema 1 already contains this gate.

But the second patch (discussed above) fixes flags in a way that could not be represented in schema1. So it is actually required.

Comment (grwlf) [reply frontend/catalyst/utils/toml.py]: David, here is the related [conversations](https://github.com/PennyLaneAI/pennylane-lightning/pull/642#discussion_r1535478642) in PL 642.  Both `C(GlobalPhase)` and `Projector` are not supported by Kokkos C++, but they are supported by its Python API.
* Projector patch is outdated because we relaxed the validation for observables: we allow devices to have Python-only observables.
* For C(GlobalPhase) we do not have a way to represent it even in schema 2. So the patch is needed and more, it has to be moved to the common execution pipeline.

Comment (dime10) [reply frontend/catalyst/utils/toml.py]: I don't think the validation is good. It is a crutch for us as developers not to forget to update the lightning toml files when their device capabilities change, but has become obsolete with the new device api (no more `.operations` class property) and causes too much confusion and issues when the Python device and C++ device diverge.

Comment (grwlf) [reply frontend/catalyst/utils/toml.py]: Well, the original purpose of validation was to solve the weird situation where one team develops APIs and another team (we) is responsible for the correct API description. If the validation is no longer possible (no .operators) - then indeed there is no use for it, but this means that we must resolve the situation in some other way.

Comment (grwlf) [reply frontend/catalyst/utils/toml.py]: David, thank you for checking this story.

**Pull request merged**
